File: Beecrowd/2 - Ad-Hoc/1026 - ~/1030 - Flavious Josephus Legend/C++/1030_FlaviousJosephusLegend.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int testCases;
    cin >> testCases;

    for (int i = 0; i < testCases; i++) {
        int peoples, steps;
        cin >> peoples >> steps;

        vector<int> people(peoples);
        for (int j = 0; j < peoples; j++) {
            people[j] = j + 1;
        }

        int idx = 0;
        while (people.size() > 1) {
            idx = (idx + steps - 1) % people.size();
            people.erase(people.begin() + idx);
        }

        cout << "Case " << i + 1 << ": " << people[0] << endl;
    }
    
    return 0;
}

File: Beecrowd/2 - Ad-Hoc/1026 - ~/1030 - Flavious Josephus Legend/1030_FlaviousJosephusLegend.py
testCases = int(input())

for i in range(testCases):
    peoples, steps = map(int, input().split())

    people = list(range(1, peoples + 1))

    idx = 0
    while len(people) > 1:
        idx = (idx + steps - 1) % len(people)
        people.pop(idx)

    print(f"Case {i + 1}: {people[0]}")
